src/sql/connection.py

```diff
@@ -110,6 +110,29 @@ def rough_dict_get(dct, sought, default=None):
     return default
 
 
+def is_pep249_compliant(conn):
+    """
+    Checks if given connection object complies with PEP 249
+    """
+    pep249_methods = [
+        "close",
+        "commit",
+        "rollback",
```

Comment:
I think `rollback()` is optional according to [PEP 249](https://peps.python.org/pep-0249/#connection-methods)
```suggestion
        # "rollback",
```

> This method is optional since not all databases provide transaction support. [[3]](https://peps.python.org/pep-0249/#id51)
In case a database does provide transactions this method causes the database to roll back to the start of any pending transaction. Closing a connection without committing the changes first will cause an implicit rollback to be performed.